src/shims/unix/macos/foreign_items.rs

```diff
@@ -164,24 +164,58 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
             // Threading
             "pthread_setname_np" => {
                 let [name] = this.check_shim(abi, Abi::C { unwind: false }, link_name, args)?;
+
+                // The real implementation has logic in two places:
+                // * in userland at https://github.com/apple-oss-distributions/libpthread/blob/c032e0b076700a0a47db75528a282b8d3a06531a/src/pthread.c#L1178-L1200,
+                // * in kernel at https://github.com/apple-oss-distributions/xnu/blob/8d741a5de7ff4191bf97d57b9f54c2f6d4a15585/bsd/kern/proc_info.c#L3218-L3227.
+                //
+                // The function in libc calls the kernel to validate
+                // the security policies and the input. If all of the requirements
+                // are met, then the name is set and 0 is returned. Otherwise, if
+                // the specified name is lomnger than MAXTHREADNAMESIZE, then
+                // ENAMETOOLONG is returned.
+                //
+                // FIXME: the real implementation maybe returns ESRCH if the thread ID is invalid.
+                // Perhaps due to the fact, that there's no man page for that function by Apple,
+                // and not all the kernel code is open sourced.
```

Comment:
I don't understand this comment. What is "perhaps due to the fact"?

Reply:
I meant that that it perhaps returns `ESRCH`, but that's not 100% true. Unfortunately, Apples OSS means here's a mirror of our repository and nothing more.

Reply:
Then just remove the comment entirely, I don't see why we'd include such speculation here.

Reply:
I will leave the FIXME still. Otherwise, there's an ICE.